transport/wsn/src/test/java/org/kaazing/gateway/transport/wsn/specification/httpx/ExtendedHandshakeIT.java

```diff
@@ -0,0 +1,139 @@
+
+    private GatewayRule gateway = new GatewayRule() {
+        {
+            GatewayConfiguration configuration =
+                    new GatewayConfigurationBuilder()
+                        .service()
+                            .accept(URI.create("ws://localhost:8080/path"))
+                            .type("echo")
+                            .crossOrigin()
+                                .allowOrigin("*")
+                            .done()
+                            .acceptOption("ws.inactivity.timeout", "123secs")
+                        .done()
+                        .service()
+                            .accept(URI.create("ws://localhost:8080/auth"))
+                            .type("echo")
+                            .realmName("Kaazing WebSocket Gateway Demo")
+                            .crossOrigin()
+                                .allowOrigin("*")
+                            .done()
+                            .authorization()
+                                .requireRole("AUTHORIZED")
+                            .done()
+                            .acceptOption("ws.inactivity.timeout", "123secs")
+                        .done()
+                        .security()
+                            .realm()
+                                .name("Kaazing WebSocket Gateway Demo")
+                                .description("Kaazing WebSocket Gateway Demo")
+                                .httpChallengeScheme("Application Basic")
+                                .authorizationMode("challenge")
+                                .loginModule()
+                                    .type("class:org.kaazing.gateway.security.auth.YesLoginModule")
+                                    .success("requisite")
+                                    .option("roles", "AUTHORIZED")
+                                .done()
+                            .done()
+                        .done()
+                    .done();
+            init(configuration);
+        }
+    };
+
+    @Rule
+    public TestRule chain = createRuleChain(gateway, k3po);
+
+    @Test
+    @Specification({ "client.sends.message.between.opening.and.extended.handshake/request" })
+    public void shouldFailWhenClientSendsMessageBetweenOpeningAndExtendedHandshake() throws Exception {
+        k3po.finish();
+    }
+
+    @Test
+    @Specification({ "connection.established/request" })
+    public void shouldEstablishConnection() throws Exception {
+        k3po.finish();
+    }
+
+    @Test
+    @Specification({ "connection.established.with.authorization/request" })
+    public void shouldEstablishConnectionWithAuthorization() throws Exception {
+        k3po.finish();
+    }
+
+    @Test
+    @Specification({ "extended.handshake.response.code.101/request"})
+    public void shouldPassWhenWebSocketProtocolGets101StatusCode() throws Exception {
+        k3po.finish();
+    }
+
+    @Test
+    @Ignore("https://github.com/kaazing/tickets/issues/314")
```

Comment:
It would be nice to update the Gateway configuration in the test so that this method can be run directly without any manual steps. See my comment in https://github.com/kaazing/tickets/issues/314. But if you don't have time we can justify that to whenever we fix that bug.